### PR TITLE
Ggoryl landmarks

### PR DESCRIPTION
**Pull request by polquant**
Incorporating data from landmarks (preprocessed mediapipe output) into model

### CODE REVIEW

File: datasets/dataset.py [hunk 1]
```diff
@@ -201,25 +203,13 @@ def _sanity_check_samples(self):
...
-            os.path.join(self.root_path, video_name, video_name + "_pose.csv"),
-            index_col=col_name,
-        ).loc[indices, :]
+    def _get_landmarks(self, video_name: str
+                       ) -> Tuple["np.ndarray[float]", "np.ndarray[float]", "np.ndarray[float]", "np.ndarray[float]"]:
+        landmarks = np.load(os.path.join(self.root_path, video_name, video_name + ".npz"))
+        face = landmarks['face']
+        left_hand = landmarks['left_hand']
+        right_hand = landmarks['right_hand']
+        pose = landmarks['pose']
```
Comment (AgaMiko) [inline datasets/dataset.py:212]: Great! Looks very neat now.

File: datasets/dataset_creator.py [hunk 2]
```diff
@@ -33,10 +33,10 @@ def get_train_and_val_subsets(self) -> (torch.utils.data.dataset.Subset, torch.u
 
         train_len, val_len = self._get_split_lens(train_dataset)
 
-        _, train_subset = torch.utils.data.random_split(train_dataset, [train_len, val_len],
+        train_subset, _ = torch.utils.data.random_split(train_dataset, [train_len, val_len],
                                                         generator=torch.Generator().manual_seed(0))
```
Comment (AgaMiko) [inline datasets/dataset_creator.py:37]: nice catch!

File: models/model.py [hunk 3]
```diff
@@ -97,11 +97,30 @@ def __init__(
...
-            representation_size = representation_size,
-            model_path=feature_extractor_model_path,
+                feature_extractor_name=feature_extractor_name,
+                representation_size=representation_size,
+                model_path=feature_extractor_model_path,
+            )
+        )
+
+        self.landmarks_model = nn.Sequential(
+            nn.LazyLinear(representation_size),
```
Comment (AgaMiko) [inline models/model.py:107]: Can you please define `landmarks_model` in a separate file? We try to keep one file per model convention

File: models/model.py [hunk 4]
```diff
@@ -97,11 +97,30 @@ def __init__(
...
+            nn.Linear(representation_size, representation_size),
+            nn.ELU(0.1),
+            nn.Dropout(transformer_dropout_rate),
+            nn.Linear(representation_size, representation_size),
+            nn.ELU(0.1),
+            nn.Dropout(transformer_dropout_rate)
         )
+
+        self.pretransformer_model = nn.Sequential(
+            nn.Linear(representation_size + representation_size, representation_size),
```
Comment (AgaMiko) [inline models/model.py:119]: Can you also define `pretransformer_model ` in a separate to keep one file per model convention

File: models/model.py [hunk 5]
```diff
@@ -123,31 +142,50 @@ def __init__(
...
+        targets, predictions, losses = self._process_batch(batch)
         if self.run:
-            self.run["metrics/batch/validation_loss"].log(loss)
-        return {"val_loss": loss, "targets": targets, "predictions": predictions}
+            self.run["metrics/batch/validation_loss"].log(losses)
+        return {"val_loss": losses, "targets": targets, "predictions": predictions}
+
+    def _process_batch(self, batch):
+        frames, targets_and_landmarks = batch
+        targets = targets_and_landmarks["target"]
```
Comment (AgaMiko) [inline models/model.py:184]: I am curious, why `targets_and_landmarks` are one variable instead of two? (`frames, targets, landmarks = batch`)

**Comment by AgaMiko** [comment]
Looks very nice! If all changes will be accepted, please use "squash and merge" option to avoid pushing >10 commits.

File: models/model.py (hunk 5 above)
Comment (polquant) [reply models/model.py]: Resolved

File: models/model.py (hunk 3 above)
Comment (polquant) [reply models/model.py]: Resolved

File: models/model.py (hunk 4 above)
Comment (polquant) [reply models/model.py]: Resolved

File: datasets/dataset_creator.py (hunk 2 above)
Comment (majsylw) [reply datasets/dataset_creator.py:37]: Ok, now I understand your comment at basecamp ;-)

File: models/features_sequential_model.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    """ Basic sequential model for preparing input to transformer """
+
+    def __init__(self, representation_size=1024, dropout_rate=0.2):
+        super().__init__()
+
+        self.representation_size = representation_size
+        self.dropout_rate = dropout_rate
+        self.model = nn.Sequential(
+            nn.LazyLinear(representation_size),
+            nn.ELU(0.1),
```
Comment (majsylw) [inline models/features_sequential_model.py:14]: why do not put this parameter (0.1) into constructor?

File: train_config.yml [hunk 7]
```diff
@@ -1,9 +1,9 @@
 augmentations:
```
Comment (majsylw) [inline train_config.yml:1]: please try do not push your config file - I usually loose mine while pulling changes :(

File: datasets/dataset.py [hunk 8]
```diff
@@ -137,6 +137,7 @@ def __init__(
         time: Union[float, None] = None,
         imagefile_template: str = "{:s}_{:d}.jpg",
         landmarks: bool = False,
+        use_face_landmarks: bool = False,
```
Comment (majsylw) [inline datasets/dataset.py:140]: do we have possibility to use face landmarks changing the parameter from train.py?
I will be good to have that option

**Review by majsylw — APPROVED** [review]

File: models/model.py (hunk 4 above)
Comment (AgaMiko) [reply models/model.py]: Looks great, thanks! I'm wondering whether we should add maybe a new folder for those models? So it's not mistaken with the main model for training? Let me know what do you think!

File: datasets/dataset.py (hunk 8 above)
Comment (polquant) [reply datasets/dataset.py:140]: Yes one only needs to use this argument when a contractor is invoked.

**Review by wiktorfilipiuk — APPROVED** [review]
Please rebase with the newest yaml file version. inputs to model_instance are different right now

**Pull request merged**